url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FisImporter.java

```diff
@@ -0,0 +1,626 @@
+            return SmallestOfMaximum.class.getSimpleName();
+        }
+        if ("wtaver".equals(name)) {
+            return WeightedAverage.class.getSimpleName();
+        }
+        if ("wtsum".equals(name)) {
+            return WeightedSum.class.getSimpleName();
+        }
+        return name;
+    }
+
+    protected Op.Pair<Double, Double> extractRange(String range) {
+        List<String> minmax = Op.split(range, " ");
+        if (minmax.size() != 2) {
+            throw new RuntimeException(String.format(
+                    "[syntax error] expected range in format '[begin end]', "
+                    + "but found <%s>", range));
+        }
+        String begin = minmax.get(0);
+        String end = minmax.get(1);
+        if (begin.charAt(0) != '[' || end.charAt(end.length() - 1) != ']') {
+            throw new RuntimeException(String.format(
+                    "[syntax error] expected range in format '[begin end]', "
+                    + "but found <%s>", range));
+        }
+        Op.Pair<Double, Double> result = new Op.Pair<>();
+        result.first = Op.toDouble(begin.substring(1));
+        result.second = Op.toDouble(end.substring(0, end.length() - 1));
+        return result;
+    }
+
+    protected Term extractTerm(String fis) {
+        String line = "";
+        for (char c : fis.toCharArray()) {
+            if (!(c == '[' || c == ']')) {
+                line += c;
+            }
+        }
+
+        List<String> nameTerm = Op.split(line, ":");
+        if (nameTerm.size() != 2) {
+            throw new RuntimeException(String.format(
+                    "[syntax error] expected term in format 'name':'class',[params], "
+                    + "but found <%s>", line));
+        }
+
+        List<String> termParams = Op.split(nameTerm.get(1), ",");
+        if (termParams.size() != 2) {
+            throw new RuntimeException(String.format(
+                    "[syntax error] expected term in format 'name':'class',[params], "
+                    + "but found <%s>", line));
+        }
+
+        List<String> parameters = Op.split(termParams.get(1), " ");
+        for (int i = 0; i < parameters.size(); ++i) {
+            parameters.set(i, parameters.get(i).trim());
+        }
+
+        return createInstance(
+                termParams.get(0).trim(),
+                nameTerm.get(0).trim(),
+                parameters);
+    }
+
+    protected Term prepareTerm(Term term, Engine engine) {
+        if (term instanceof Linear) {
+            Linear linear = (Linear) term;
+            linear.set(linear.coefficients, engine.getInputVariables());
+        } else if (term instanceof Function) {
+            Function function = (Function) term;
+            function.setEngine(engine);
+            //builtin functions are loaded from TermFactory
+            function.load();
+        }
+        return term;
+    }
+
+    protected Term createInstance(String mClass, String name, List<String> parameters) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 585-610, lines 612-624 into separate methods.